data-products/src/sql_etl/sql/glean_firefox_new_tab_impressions_daily/glean_firefox_new_tab_daily_unique_engagement_by_feed/data.sql

```diff
@@ -0,0 +1,131 @@
+      WHEN ( normalized_country_code IN ('GB',
+        'IE')
+      AND metrics.string.newtab_locale IN ('en-CA',
+        'en-GB',
+        'en-US') ) THEN 'NEW_TAB_EN_GB'
+      WHEN ( normalized_country_code IN ('IN') AND metrics.string.newtab_locale IN ('en-CA', 'en-GB', 'en-US') ) THEN 'NEW_TAB_EN_INTL'
+      WHEN ( normalized_country_code IN ('DE',
+        'CH',
+        'AT',
+        'BE')
+      AND metrics.string.newtab_locale IN ('de',
+        'de-AT',
+        'de-CH') ) THEN 'NEW_TAB_DE_DE'
+      WHEN (normalized_country_code IN ('IT') AND metrics.string.newtab_locale IN ('it')) THEN 'NEW_TAB_IT_IT'
+      WHEN (normalized_country_code IN ('FR')
+      AND metrics.string.newtab_locale IN ('fr')) THEN 'NEW_TAB_FR_FR'
+      WHEN (normalized_country_code IN ('ES') AND metrics.string.newtab_locale IN ('es-ES')) THEN 'NEW_TAB_ES_ES'
+  END
+    AS feed_name,
+    e.name AS event_name,
+    SAFE_CAST(mozfun.map.get_key(e.extra,
+        'is_sponsored') AS boolean) AS is_sponsored,
+    metrics.boolean.pocket_enabled = TRUE AS pocket_enabled,
+    metrics.boolean.pocket_sponsored_stories_enabled = TRUE AS pocket_sponsored_stories_enabled,
+    client_info.client_id AS client_id
+  FROM
+    deduplicated_pings,
+    UNNEST(events) AS e
+    --filter to Pocket events
+  WHERE
+    e.category = 'pocket'
+    --limit to impressions and clicks
+    AND e.name IN ('impression',
+      'click')
+    --keep only data with a non-null recommendation ID or tile ID
+    AND (mozfun.map.get_key(e.extra,
+        'recommendation_id') IS NOT NULL
+      OR mozfun.map.get_key(e.extra,
+        'tile_id') IS NOT NULL)
+    --include only release channel data
+    AND normalized_channel = 'release'
+    --include only data from Firefox 121+
+    AND SAFE_CAST(SPLIT(client_info.app_display_version, '.')[0] AS int64) >= 121
+    --limit to locale & country combinations with Pocket enabled
+    AND ( ( normalized_country_code IN ('US',
+          'CA',
+          'GB',
+          'IE',
+          'IN')
+        AND metrics.string.newtab_locale IN ('en-CA',
+          'en-GB',
+          'en-US') )
+      OR ( normalized_country_code IN ('DE',
+          'CH',
+          'AT',
+          'BE')
+        AND metrics.string.newtab_locale IN ('de',
+          'de-AT',
+          'de-CH') )
+      OR (normalized_country_code IN ('IT')
+        AND metrics.string.newtab_locale IN ('it'))
+      OR (normalized_country_code IN ('FR')
+        AND metrics.string.newtab_locale IN ('fr'))
+      OR (normalized_country_code IN ('ES')
+        AND metrics.string.newtab_locale IN ('es-ES')) ) )
+  --let's skip the usual click deduplication step, Jeff Silverman's analysis showed that Glean is resistant to
+  --the duplicate clicks issue we saw with the legacy PingCentre events:
+  --https://docs.google.com/document/d/1aL3bjJ6PQLHH455zCMReDpYnyqPTNwTvTW2QujwP-nw/edit#heading=h.bz9b6bjjzei
```

Comment:
I'm confused. The final query uses `COUNT DISTINCT` on `client_id` for all aggregate columns. Doesn't that mean we are deduplicating?

Reply:
Yes, you're right. Sorry, I think this comment is in here by accident. @chelseybeck feel free to remove it!